be/src/vec/functions/array/function_array_set.h

```diff
@@ -181,7 +181,11 @@ struct ArraySetImpl {
         constexpr auto execute_left_column_first = Impl::Action::execute_left_column_first;
         size_t current = 0;
         Impl impl;
-        for (size_t row = 0; row < left_data.offsets_ptr->size(); ++row) {
+        size_t row_size = left_data.offsets_ptr->size();
```

Comment:
what if right const

Reply:
right and left is all not const ==>  left_data.offsets_ptr->size()
right const && left is not const ==> left_data.offsets_ptr->size()
right is not const && left is const ==> right.offsets_ptr->size()
right and left is all const ==>  left_data.offsets_ptr->size()